packages/language/src/workspace/compilation-unit.ts

```diff
@@ -177,15 +178,23 @@ export class CompilationUnitHandler {
   }
 
   /**
-   * Gets an existing or creates a new compilation unit for the given URI
+   * Gets an existing or creates a new compilation unit for the given URI, except for standalone library files
    *
-   * @returns Pre-existing or new compilation unit
+   * @returns Pre-existing or new compilation unit, or undefined if it's a standalone library file
    */
-  getOrCreateCompilationUnit(uri: URI): CompilationUnit {
-    return (
-      this.compilationUnits.get(uri.toString()) ||
-      this.createAndStoreCompilationUnit(uri)
-    );
+  getOrCreateCompilationUnit(uri: URI): CompilationUnit | undefined {
+    if (this.compilationUnits.has(uri.toString())) {
+      // existing compilation unit
+      return this.compilationUnits.get(uri.toString());
+    } else if (
+      !PluginConfigurationProviderInstance.isLibFileCandidate(uri.toString())
```

Comment:
We need to skip the encoding here, otherwise the `c:/` on windows gets encoded into `c%3A/`

```suggestion
      !PluginConfigurationProviderInstance.isLibFileCandidate(uri.toString(true))
```